crates/turborepo-filewatch/src/package_watcher.rs

```diff
@@ -1040,4 +1025,200 @@ mod test {
             ]
         );
     }
+
+    #[tokio::test]
+    #[tracing_test::traced_test]
+    async fn pnpm_invalid_states_test() {
+        let tmp = tempfile::tempdir().unwrap();
+        let repo_root = AbsoluteSystemPathBuf::try_from(tmp.path())
+            .unwrap()
+            .to_realpath()
+            .unwrap();
+
+        let workspaces_path = repo_root.join_component("pnpm-workspace.yaml");
+        // Currently we require valid state to start the daemon
+        let root_package_json_path = repo_root.join_component("package.json");
+        // Start with no workspace glob
+        root_package_json_path
+            .create_with_contents(r#"{"packageManager": "pnpm@7.0"}"#)
+            .unwrap();
+        repo_root
+            .join_component("pnpm-lock.yaml")
+            .create_with_contents("")
+            .unwrap();
+
+        let watcher = FileSystemWatcher::new_with_default_cookie_dir(&repo_root).unwrap();
+        let recv = watcher.watch();
+        let cookie_writer = CookieWriter::new(
+            watcher.cookie_dir(),
+            Duration::from_millis(100),
+            recv.clone(),
+        );
+
+        let package_watcher = PackageWatcher::new(
+            repo_root.clone(),
+            recv,
+            LocalPackageDiscoveryBuilder::new(repo_root.clone(), None, None)
+                .build()
+                .unwrap(),
+            cookie_writer,
+        )
+        .unwrap();
+
+        // TODO: change this to expect either an empty result, or a package manager
+        // without globs
+        tokio::time::timeout(
+            Duration::from_millis(50),
```

Comment:
Until we address the TODO, if we want to verify that this is truly waiting, we might need to bump up the timeout. If it can take up to 200ms to get a result, I feel that the 50ms might low enough to give a false negative.

Reply:
I'm intending on addressing the TODO before flagging any of this code on. Roughly 2 PRs from now we'll return negative answers, rather than relying on timeouts.